apps/aecore/lib/aecore/structures/oracle_response_tx_data.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Aecore.Structures.OracleResponseTxData do
+  alias __MODULE__
+  alias Aecore.Oracle.Oracle
+  alias Aecore.Chain.Worker, as: Chain
+  alias Aecore.Wallet.Worker, as: Wallet
+  alias Aecore.Chain.ChainState
+  alias Aecore.Structures.Account
+
+  require Logger
+
+  @type tx_type_state :: ChainState.oracles()
+
+  @type payload :: %{
+          query_id: binary(),
+          response: map()
+        }
+
+  @type t :: %OracleResponseTxData{
+          query_id: binary(),
+          response: map()
+        }
+
+  defstruct [:query_id, :response]
+  use ExConstructor
+
+  @spec get_chain_state_name() :: :oracles
+  def get_chain_state_name(), do: :oracles
+
+  @spec init(payload()) :: OracleResponseTxData.t()
+  def init(%{
+        query_id: query_id,
+        response: response
+      }) do
+    %OracleResponseTxData{
+      query_id: query_id,
+      response: response
+    }
+  end
+
+  @spec is_valid?(OracleResponseTxData.t()) :: boolean()
+  def is_valid?(%OracleResponseTxData{}) do
+    true
+  end
+
+  @spec process_chainstate!(
+          OracleResponseTxData.t(),
+          Wallet.pubkey(),
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          ChainState.account(),
+          tx_type_state()
+        ) :: {ChainState.accounts(), tx_type_state()}
+  def process_chainstate!(
+        %OracleResponseTxData{} = tx,
+        sender,
+        fee,
+        nonce,
+        block_height,
+        accounts,
+        %{interaction_objects: interaction_objects} = oracle_state
+      ) do
+    case preprocess_check(
+           tx,
+           sender,
+           Map.get(accounts, sender, Account.empty()),
+           fee,
+           nonce,
+           block_height,
+           oracle_state
+         ) do
+      :ok ->
+        interaction_object = interaction_objects[tx.query_id]
+        query_fee = interaction_object.query.query_fee
+
+        new_senderount_state =
+          Map.get(accounts, sender, Account.empty())
+          |> deduct_fee(fee - query_fee)
+
+        updated_accounts_chainstate = Map.put(accounts, sender, new_senderount_state)
+
+        updated_interaction_objects =
+          Map.put(interaction_objects, tx.query_id, %{
+            interaction_object
+            | response: tx,
```

Comment:
shouldn't it be `| response: tx.response,`?

Reply:
yes that will be better